Build warnings with TF-M that are not treated as errors

### Describe the bug

when building anything with TF-M, there are build warnings that are not being treated as errors, so there are 2 issues:

- The build warnings themselves
- The fact that -Werror is not being propagted to TF-M and those are not detected in CI and during development

### Regression

- [ ] This is a regression.

### Steps to reproduce

` west   build -p  -b  mps2/an521/cpu0/ns  samples/hello_world`

### Relevant log output

```shell
[48/378] Building C object bl2/ext/mcuboot/bootutil/CMakeFiles/bootutil.dir/src/bootutil_misc.o
/home/nashif/zephyrproject/bootloader/mcuboot/boot/bootutil/src/bootutil_misc.c: In function 'boot_trailer_scramble_offset':
/home/nashif/zephyrproject/bootloader/mcuboot/boot/bootutil/src/bootutil_misc.c:153:15: warning: implicit declaration of function 'flash_area_get_sector'; did you mean 'flash_area_get_sectors'? [-Wimplicit-function-declaration]
  153 |         ret = flash_area_get_sector(fa, flash_area_get_size(fa) - boot_trailer_sz(alignment),
      |               ^~~~~~~~~~~~~~~~~~~~~
      |               flash_area_get_sectors
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

Zephyr SDK 0.17.1 on Linux on latest main branch v4.1.0-4985-ga098f5c3a39

### Additional Context

_No response_

--- a/modules/trusted-firmware-m/CMakeLists.txt
+++ b/modules/trusted-firmware-m/CMakeLists.txt
@@ -26,6 +26,11 @@ if (CONFIG_BUILD_WITH_TFM)
     endif()
   endif()
 
+  # Treat any warning as error
+  if (CONFIG_COMPILER_WARNINGS_AS_ERRORS)
+    list(APPEND TFM_CMAKE_ARGS -DCONFIG_TFM_WARNINGS_ARE_ERRORS:BOOL=ON)
+  endif()
+
   if (CONFIG_TFM_SFN)
     list(APPEND TFM_CMAKE_ARGS -DCONFIG_TFM_SPM_BACKEND="SFN")
   else() # CONFIG_TFM_IPC

--- a/west.yml
+++ b/west.yml
@@ -360,7 +360,7 @@ manifest:
       groups:
         - tee
     - name: trusted-firmware-m
-      revision: d38260c937f957e4617691daebc7b565e82a84c4
+      revision: 021e2bbd50c215e41710a72e05abce3224f074a7
       path: modules/tee/tf-m/trusted-firmware-m
       groups:
         - tee

